### DISCUSSION TITLE
Training duration and GPU setup for tacotron2 sota models

### DISCUSSION THREAD

**Discussion by padmalcom** [speechbrain/speechbrain#1757] [General]
Hi, I'm training a German tacotron2 tts model on 6 gpus with 24gb memory each using data parallel in a custom 30gb training dataset. 

Training one epoch takes about 10h (from checkpoint to checkpoint) with a batch size of 14. So training 750 epochs as proposed in the hyper parameters will take ~73 days.

Since this is a lot of time I wonder what your setup was (how many gpus with how much memory? Size of your dataset? How many days did you train?).

Thanks,
Jonas

**Comment by BenoitWang** [comment]
Hi @padmalcom, that seems too long, maybe check if something is running on cpu.

I ran Tacotron2 on LJSpeech(24 hours in total) with 1*v100, batchsize 64 and that takes about 10 minutes per epoch.

And please use ddp instead of dp because dp is causing issues...

**Reply by padmalcom** [reply]
Hi @BenoitWang, thank you, that confirms what I thought. Unfortunately, we have to use bitfusion in our virtualized setup which is really hard to debug and it is nearly impossible to understand what causes the performance issues. I checked that cuda is available at each point in the training process but I fear I have to check the vm layer with our operations team.

**Comment by padmalcom** [comment]
@BenoitWang Do you remember the seconds/iteration you got during the training process you mentioned?
